### PR TITLE
Decorate `RelativisticBody` and `relativistic_energy` with `@particle_input`

### PR DESCRIPTION
**Pull request by namurphy**
## Description

- Applied `particle_input` to `relativistic_energy` and `RelativisticBody`
- Updated docstrings
- Changed the parameters provided to `relativistic_energy`:
  - Changed `m` to `particle`
  - Changed `v` to `V`
  This would be a breaking change; however, when provided as positional arguments, the change would be transparent.  In particular, `particle_input` would create a `CustomParticle` from a `Quantity` for mass.

## Motivation and context

The goals here are
 - To further implement `particle_input` in `plasmapy.formulary` for more compatibility with `ParticleList` and `CustomParticle` objects.
 - To improve consistency with function signatures.  For example, we mostly use `V` instead of `v` to describe velocity, and it helps to be consistent.

I'm wondering if it would be worth having a deprecation warning for this and still accept the old parameters as keywords, but `relativistic_energy` might not be widely used enough to warrant the extra complexity this would entail.

## Related issues

This is approximately 4% of #341!

### CODE REVIEW

File: plasmapy/formulary/relativity.py [hunk 1]
```diff
@@ -93,31 +94,57 @@ def Lorentz_factor(V: u.m / u.s):
...
+    where :math:`m` is the mass of the body and :math:`γ` is its
+    `~plasmapy.formulary.relativity.Lorentz_factor`.
 
     Parameters
     ----------
-    m : `~astropy.units.Quantity`
-        The mass in units convertible to kilograms.
+    particle : |particle-like|
+        A representation of a particle or a `~astropy.units.Quantity`
+        with units of mass.
```
Comment (namurphy) [inline plasmapy/formulary/relativity.py:122]: Still need to expand this.  We really need to standardize the parameter descriptions for particle-like inputs...

**Comment by namurphy** [comment]
Looks like the particle factory function isn't accepting `Quantity` objects that are arrays with physical type mass or charge.  Will need to fix that before tests will pass in this PR.

**Comment by StanczakDominik** [comment]
> but relativistic_energy might not be widely used enough to warrant the extra complexity this would entail.

Might not, or it might! You can't really assume that. I think if we're breaking stuff here, we need the deprecationwarning.

File: plasmapy/formulary/relativity.py [hunk 2]
```diff
@@ -93,46 +94,56 @@ def Lorentz_factor(V: u.m / u.s):
...
+    The total energy of a relativistic body is:
 
     .. math::
 
-        E = γ m c^2
+        E = γ m c^2,
 
-    where :math:`γ` is the `Lorentz_factor`. This function returns the
-    sum of the mass energy and the kinetic energy.
+    where :math:`m` is the mass of the body and :math:`γ` is its
```
[the remark this answers is not in the document]
Comment (namurphy) [reply plasmapy/formulary/relativity.py:115]: Good catch!  I thought I had made this change, but that might have just been in a dream or something.

**Review by ejohnson-96 — APPROVED** [review]
:)

**Pull request merged**